components/benefit_sponsors/app/controllers/benefit_sponsors/profiles/broker_agencies/broker_agency_staff_roles_controller.rb

```diff
@@ -7,15 +7,20 @@ module BrokerAgencies
       class BrokerAgencyStaffRolesController < ::BenefitSponsors::ApplicationController
 
         def new
-          profile = find_broker_agency_profile
-          authorize profile
+          # this endpoint is used for two scenarios
+          # 1.) to render the form for brokers/agents creating new broker agency staff roles for their broker agencies
+          # 2.) to render a different form in the view for benefit_sponsors/profiles/registrations/new for non-users sending in applications to be staff for existing broker agencies
+          # authorization is only enforced here for the first scenario
+          authorize @broker_agency_profile if find_broker_agency_profile
```

Comment:
- user is still able to bypass authorization with fake bson id.
- Requested for additional changes

Reply:
changed method to a before action, requested changes made!